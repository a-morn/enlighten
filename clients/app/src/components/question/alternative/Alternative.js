import React, { memo } from 'react'
import './Alternative.scss'

const Alternative = memo(
  ({
    onClick,
    selected,
    correct,
    disabled,
    alternative: { text, src, type },
  }) => {
    const classNames = [
      selected && 'selected',
      correct && 'correct',
      correct === false && 'incorrect',
      disabled && 'opacity-50 cursor-not-allowed',
      `alternative--${type}`,
    ]
      .filter(cn => cn)
      .join(' ')
    return (
      <button
        className={`w-full bg-gray-300 hover:bg-gray-400 text-gray-800 py-2 px-4 rounded bg-origin-content ${classNames}`}
        onClick={onClick}
        disabled={disabled}
        style={{ backgroundImage: `url(${src})` }}
      >
        {type === 'text' && text}
      </button>
    )
  },
)

export default Alternative
