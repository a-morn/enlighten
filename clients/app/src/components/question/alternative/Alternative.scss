.alternative {
	button {
		min-height: 50px;
		&.incorrect.selected {
			animation-name: bg-color-pulse-incorrect;
		}
		&.correct:not(.selected) {
			animation-name: bg-color-pulse-correct;
		}
		&.correct.selected {
			background-color: green;
		}

		animation-duration: 1s;
		animation-iteration-count: infinite;

		&.alternative--image {
			background-size: contain;
			background-repeat: no-repeat;
			background-position: center center;
			background-origin: border-box
		}
	}
}

@keyframes bg-color-pulse-incorrect {
	0% {
		background-color: red;
	}
	100% {
		background-color: darkred;
	}
}

@keyframes bg-color-pulse-correct {
	0% {
		background-color: green;
	}
	100% {
		background-color: darkgreen;
	}
}
