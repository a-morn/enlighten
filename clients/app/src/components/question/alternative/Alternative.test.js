import React from 'react'
import { render } from '@testing-library/react'
import Alternative from './Alternative'

test("doesn't have classes correct, incorrect, or selected by default", () => {
  const { getByTestId } = render(<Alternative alternative={{ type: 'text' }} />)

  expect(getByTestId('alternative-button').classList.contains('correct')).toBe(
    false,
  )
  expect(getByTestId('alternative-button').classList.contains('selected')).toBe(
    false,
  )
  expect(
    getByTestId('alternative-button').classList.contains('incorrect'),
  ).toBe(false)
})

test('has class correct if @correct == true , and class selected if @selected == true', () => {
  const { getByTestId } = render(
    <Alternative alternative={{ type: 'text' }} selected correct />,
  )

  expect(getByTestId('alternative-button').classList.contains('correct')).toBe(
    true,
  )
  expect(getByTestId('alternative-button').classList.contains('selected')).toBe(
    true,
  )
})

test('has class incorrect if correct === false', () => {
  const { getByTestId } = render(
    <Alternative alternative={{ type: 'text' }} correct={false} />,
  )

  expect(
    getByTestId('alternative-button').classList.contains('incorrect'),
  ).toBe(true)
})
