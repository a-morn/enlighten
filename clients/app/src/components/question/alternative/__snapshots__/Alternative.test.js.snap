// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alternative component snapshot matches the snapshot 1`] = `
<button
  className="alternative-type-undefined"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
/>
`;
